robottelo/utils/datafactory.py

```diff
@@ -367,6 +367,24 @@ def valid_hostgroups_list():
     ]
 
 
+@filtered_datapoint
+def valid_hostgroups_list_short():
+    """Generates a list of valid host group names. Shorter so they can be nested.
```

Comment:
I am not quite sold on this - you declare its properties quite vaguely ( _shorter_), but the reality is, you have a test dependent on this that requires this to be no shorter than 8 items.
If someone decides that it is not short enough and decides to make it even shorter, your new test would start to raise index errors.

I'd suggest to ditch this and simply generate your dataset in a local fixture.

Reply:
`short` stands for a `list of short names`, not `short list of names`. There is a similar fixture that has longer names, i.e. greater value in `randint`.
There are also other tests that require a fixture to have at least X members, `valid_hostgroups_list` being no exception.